core/src/main/scala/epollcat/internal/ch/SocketHelpers.scala

```diff
@@ -20,24 +20,32 @@ import java.io.IOException
 import java.net.InetAddress
 import java.net.InetSocketAddress
 import java.net.SocketAddress
+import java.nio.channels.UnsupportedAddressTypeException
 import scala.scalanative.libc.errno
 import scala.scalanative.meta.LinktimeInfo
 import scala.scalanative.posix
+import scala.scalanative.posix.netdbOps._
 import scala.scalanative.posix.netinet.inOps._
 import scala.scalanative.unsafe._
 
 private[ch] object SocketHelpers {
 
+  lazy val preferIPv4Stack =
+    java.lang.Boolean.parseBoolean(System.getProperty("java.net.preferIPv4Stack", "false"))
+
```

Comment:
Does parseBoolean do a case-blind comparison?
The java docs are pretty unclear about if the property is
Boolean or String.  They say
However, in the case an application would rather use IPv4 only sockets, then this property can be set to true.  "true" is bolded
but not in double quotes.  

You are probably correct here and I should fix in SN, with
your approval.

Reply:
Good question. Here's what the Javadoc says.

> Parses the string argument as a boolean. The `boolean` returned represents the value `true` if the string argument is not `null` and is equal, ignoring case, to the string `"true"`.

> Example: `Boolean.parseBoolean("True")` returns `true`.
> Example: `Boolean.parseBoolean("yes")` returns `false`.

https://docs.oracle.com/javase/8/docs/api/java/lang/Boolean.html#parseBoolean-java.lang.String-

Reply:
Good to know, thank you for the snippet.  So, presumably 0 and 1 would both be Boolean:False?
Sometimes on unix command lines (perhaps bash?)   those are also taken as boolean. There
are a bunch of quirks about the case of true/false, where something work and others (True?) 
that one would expect to work do not.  Like black flies in Maine, just an annoyance one
learn to live with (or pretend to live with).

Reply:
> Probably needs a placeholder that java.net.preferIPv6Addresses is known, but not implemented this evolution.

I am a little confused what we would do with `preferIPv6Addresses`. It seems like this is a useful option for `getByName`, which has the choice to resolve either an IPv4 address or an IPv6 address.

But in our case, we are directly handed an IP address and we have already created a socket according to the value of `preferIPv4Stack`. So it's not obvious to me how we should modify our behavior based on this setting.

Reply:
I kind of punted `preferIPv6Addresses` in SN 0.5.0 and realized doing this work that I had forgotten that
I had done so. Oops!

You can see why I moved `peferIPv6Addresses` to the rear echelons.  It is a complex feature which
will probably never be used but some reasonable answer must be made for API compatibility.

I will have to think about and do some design studies on this one.  I think the place where
I would start is if a host/machine has both IPv4 and IPv6 addresses, preferIPv6Addresses, is
set, and an IPv4 address is presented to getadderinfo. The "pure" IPv6 address should be
used. I just do not know if addrinfo will order the list that way. I suspect it must be done
manually.  I believe this a live concern for SN.  I am not convinced that such an IPv4
can sneak thru to epoll getadderinfo.  I suspect it can.

Again, when this becomes our biggest & most urgent problem, it will be a good day.